traitlets/config/application.py

```diff
@@ -128,15 +128,20 @@ class Application(SingletonConfigurable):
     # A sequence of Configurable subclasses whose config=True attributes will
     # be exposed at the command line.
     classes = []
-    @property
-    def _help_classes(self):
-        """Define `App.help_classes` if CLI classes should differ from config file classes"""
-        return getattr(self, 'help_classes', self.classes)
-    
-    @property
-    def _config_classes(self):
-        """Define `App.config_classes` if config file classes should differ from CLI classes."""
-        return getattr(self, 'config_classes', self.classes)
+
+    def _classes_inc_parents(self):
+        """Iterate through configurable classes, including configurable parents
+
+        Children should always be after parents, and each class should only be
+        yielded once.
+        """
+        seen = set()
+        for c in self.classes:
```

Comment:
classes is a list, and I think we do use the order already in that list to group things meaningfully.